File: typescript/creational/abstract_factory/sport_brand/factory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Nike = exports.Adidas = void 0;
const product_1 = require("./product");
class Adidas {
    makeShoe() {
        return new product_1.AdidasShoe();
    }
    makeShirt() {
        return new product_1.AdidasShirt();
    }
}
exports.Adidas = Adidas;
class Nike {
    makeShoe() {
        return new product_1.NikeShoe();
    }
    makeShirt() {
        return new product_1.NikeShirt();
    }
}
exports.Nike = Nike;

File: dart/creational/abstract_factory/sport_brand/bin/factory.dart
import 'product.dart';

abstract class Factory {
  Shoe makeShoe();
  Shirt makeShirt();
}

class AdidasFactory implements Factory {
  @override
  Shoe makeShoe() {
    return AdidasShoe();
  }

  @override
  Shirt makeShirt() {
    return AdidasShirt();
  }
}

class NikeFactory implements Factory {
  @override
  Shoe makeShoe() {
    return NikeShoe();
  }

  @override
  Shirt makeShirt() {
    return NikeShirt();
  }
}
